### PR TITLE
[TagBundle] add tag removal functionality from resources

### PR DESCRIPTION
**Pull request by nicolasjacquemet**
also implemented rights check for tag creation and removal

### CODE REVIEW

File: plugin/tag/Manager/TagManager.php [hunk 1]
```diff
@@ -185,6 +185,14 @@ public function getTaggedWorkspacesByRoles(
             array();
     }
 
+    public function removeTaggedObjectsByResourceAndTag($resourceNode, $tag)
+    {
+        $taggedObject = $this->getOneTaggedObjectByTagAndObject(
+                $tag, $resourceNode->getId(),
+                str_replace('Proxies\\__CG__\\', '', get_class($resourceNode)));
```
Comment (Elorfin) [inline plugin/tag/Manager/TagManager.php]: It's simplier to use `$resourceNode->getClass()`.

File: plugin/tag/Manager/TagManager.php [hunk 2]
```diff
@@ -185,6 +185,14 @@ public function getTaggedWorkspacesByRoles(
             array();
     }
 
+    public function removeTaggedObjectsByResourceAndTag($resourceNode, $tag)
```
Comment (Elorfin) [inline plugin/tag/Manager/TagManager.php]: Missing type hinting.

File: plugin/tag/Controller/TagController.php [hunk 3]
```diff
@@ -96,6 +98,10 @@ public function resourceTagFormAction(ResourceNode $resourceNode)
      */
     public function resourceTagAction(ResourceNode $resourceNode)
     {
+        if (!$this->container->get('security.authorization_checker')->isGranted('EDIT', $resourceNode)) {
```
Comment (stefk) [inline plugin/tag/Controller/TagController.php]: AFAIK resource rights checks must be done on a `ResourceCollection` instance, not directly on a resource node. Can you confirm @ngodfraind?

File: plugin/tag/Manager/TagManager.php (hunk 1 above)
Comment (nicolasjacquemet) [inline plugin/tag/Manager/TagManager.php]: @Elorfin $resourceNode->getClass() would return the underlying type (eg. blog, tag, etc), here the actual class type ResourceNode is expected.

**Comment by Elorfin** [comment]
LGTM

**Comment by stefk** [comment]
LGTM

**Pull request merged**